src/App/DataSource/Twilio/TwilioController.php

```diff
@@ -32,9 +32,26 @@ public function handleRequest(Request $request)
             abort(403, 'Incorrect or missing AccountSid');
         }
 
+        // Detect contact type
+        $to = $request->input('To');
+        $from = $request->input('From');
+        if (strpos($from, ':')) {
```

Comment:
I think if you do a strpos for whatsapp directly you can avoid the switch no?

Reply:
sure! I'd rather leave it though. My intention here was to leave it right open for other channel implementations, since it seems Twilio may be having more of those. So the first `i`f detects the presence of a channel marker and the `switch` will match the channel.